runhouse/resources/functions/aws_lambda.py

```diff
@@ -305,8 +311,17 @@ def arguments_validation(
         cls, paths_to_code, env, runtime, timeout, memory_size, tmp_size, retention_time
     ):
 
-        if isinstance(env, str) and "requirements.txt" in env:
-            paths_to_code.append(Path(env).absolute())
+        if (
+            "./" in env.reqs
+            and env.reqs[0] != "./"
+            and env.reqs[0].split("/")[0].split()[0] != env.reqs[0].split()[0]
+        ):
+            reqs = env.reqs
+            path_to_txt = str(
+                Path(reqs[0].split(":")[1] + "requirements.txt").absolute()
+            )
+            paths_to_code.append(path_to_txt)
+            env.reqs = [path_to_txt] + env.reqs[1:]
```

Comment:
Not totally clear on what's going on here, but looks pretty specific to one structure/use case. I think it would be a lot more robust if we could just send the packages to the lambda into the home directory, as we do with regular clusters. We can't rsync into a lambda, but can either 1) send the necessary files over the zip interface or 2) zip them ourselves and send them via a call into the lambda as we do with the env. I don't think that should be in this PR but feel it's quite critical.